File: Arrays/conflictingAppointments.py
def can_attend_all_appointments(intervals):
  # Sort the intervals
  intervals.sort(key=lambda x: x[0])
  start, end = 0, 1

  for i in range(1, len(intervals)):
    # Check whether two of the intervals overlap
    if intervals[i][start] < intervals[i - 1][end]:
      return False
  return True


def main():
  print("Can attend all appointments: " + str(can_attend_all_appointments([[1, 4], [2, 5], [7, 9]])))
  print("Can attend all appointments: " + str(can_attend_all_appointments([[6, 7], [2, 4], [8, 12]])))
  print("Can attend all appointments: " + str(can_attend_all_appointments([[4, 5], [2, 3], [3, 6]])))


main()

File: Arrays/ConflictingAppointments.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

record Interval(int start, int end) {}
public class ConflictingAppointments {
    public static boolean canAttendAllAppointments(Interval[] intervals) {
    List<Interval> allIntervals = 
        new ArrayList<>(Arrays.stream(intervals).toList());

    Collections.sort(allIntervals, (a, b) -> a.start() - b.start());
    Interval prev = new Interval(-1, -1);
    for(Interval interval: allIntervals) {
      if(interval.start() < prev.end()) return false;

      prev = interval;
    }

    return true;
  }

  public static void main(String[] args) {
    Interval[] intervals = new Interval[] {
      new Interval(1, 4),
      new Interval(2, 5),
      new Interval(7, 9)
    };
    System.out.println("Can attend all appointments: " + canAttendAllAppointments(intervals));
  }
}
